massa-execution-worker/src/context.rs

```diff
@@ -635,6 +635,54 @@ impl ExecutionContext {
         )
     }
 
+    /// Try to slash 1 roll from the denounced address.
+    ///
+    /// # Arguments
+    /// * `denounced_addr`: address to slash the roll or deferred credit from
+    pub fn try_slash_roll(
+        &mut self,
+        denounced_addr: &Address,
+        roll_count: u64,
+    ) -> Result<Amount, ExecutionError>
+    {
+        let slashed_coins;
+
+        // First try to slash roll if any
+        if self.speculative_roll_state.try_burn_rolls(
+            denounced_addr,
+            roll_count,
+        ).is_err() {
+
+            // if no roll were slashed, that can likely be done from deferred credit
+            let credits = self.speculative_roll_state.get_address_deferred_credits(denounced_addr, self.slot);
+            let amount_to_slash = self.config.roll_price
+                .checked_mul_u64(roll_count)
+                .ok_or_else(||
+                    ExecutionError::RuntimeError(format!("Cannot mult roll price by {}", roll_count))
+                )?;
+
+            // Remove deferred credits starting at self.slot until
+            let mut amount_to_rm = amount_to_slash;
+            for (slot, _amount) in credits.iter() {
```

Comment:
skip the slot == self.slot iteration, as explained above

Reply:
done